tools/preseal/sign.c

```diff
@@ -0,0 +1,137 @@
+#include <stdio.h>
+#include <stdint.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <wolfssl/options.h>
+#include <wolfssl/wolfcrypt/settings.h>
+#include <wolfssl/wolfcrypt/random.h>
+#include <wolfssl/wolfcrypt/sha256.h>
+#include <wolfssl/wolfcrypt/hash.h>
+#include <wolfssl/wolfcrypt/ecc.h>
+
+#define ECC_KEY_SIZE 32
+
+void usage(void) {
+    printf("Sign an image digest with an ecc private key:\n");
+    printf("./sign private-key digest");
+}
+
+int main(int argc, char** argv)
+{
+    int rc;
+    int fd;
+    uint8_t eccBuff[256];
```

Comment:
This should be sized 3*ECC_KEY_SIZE, no?  Then the check below could ensure the private key was the correct size.